executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -89,18 +94,23 @@ instance STS CHAIN where
                  }
     ]
 
-
   transitionRules = [
     do
       TRC ( Env { maximumBlockSize, participants }
           , St { currentSlot, transactionsSt }
-          , block@Block{ slot, transactions }) <- judgmentContext
+          , block@Block{ slot, transactions }
+          ) <- judgmentContext
       currentSlot < slot
         ?! BlockSlotNotIncreasing (CurrentSlot currentSlot) slot
       size block < maximumBlockSize
         ?! MaximumBlockSizeExceeded (size block) (Threshold maximumBlockSize)
+      -- TODO: we will need a header transition as well, where the votes are
+      -- tallied.
```

Comment:
We need to discuss this further in order to understand better, what is the need for the headers?

Reply:
The slot number is in the block headers. In Byron and Shelley we perform slot ticks transitions in separate header rules. In these transitions we perform computations that are dependent on slot numbers like delegation activation, or protocol parameter changes. For the new update mechanism we should do something similar.

We can discuss this over a video call for sure!